optuna/trial.py

```diff
@@ -372,14 +372,14 @@ def should_prune(self, step):
 
         Args:
             step:
-                Step of the trial (e.g., epoch of neural network training).
+                Deprecated: Step of the trial (e.g., epoch of neural network training).
```

Comment:
It might be better to call DeprecationWarning via `warnings.warn()`

Reply:
> It might be better to call `warnings.warn()` with DeprecationWarning.

Sounds good.